internal/backend/state.go

```diff
@@ -289,6 +289,11 @@ func (state *State) Close(ctx context.Context) error {
 	return state.closeState(ctx, state)
 }
 
+func (state *State) Logout(ctx context.Context) error {
```

Comment:
I don't like the fact that this is only called on logout. Clients sometimes just drop the connection (it's permitted by RFC); actually sending a logout command before disconnecting is just a nicety. 

I'd suggest we pruned the deleted messages whenever a session is closed.

Reply:
Addressed. Moved this section to `state.close()`